src/remote_common.c

```diff
@@ -0,0 +1,325 @@
+                       amqp_empty_bytes, 0, 0, 0, amqp_empty_table);
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn),
+                           "Set the consuming queue"))
+        return false;
+
+    return true;
+}
+
+bool consume_message(amqp_connection_state_t *conn,
+                     amqp_channel_t channel,
+                     amqp_envelope_t *envelope)
+{
+    amqp_maybe_release_buffers(*conn);
+    /* Consume a message from the queue in the rabbitmq broker */
+    if (!die_on_amqp_error(amqp_consume_message(*conn, envelope, NULL, 0),
+                           "Consuming the message"))
+        return false;
+
+    return true;
+}
+
+bool wait_response_message(amqp_connection_state_t *conn,
+                           amqp_channel_t channel,
+                           amqp_bytes_t callback_queue,
+                           char *frame_body,
+                           int body_len)
+{
+    amqp_frame_t frame;
+#if defined(ENABLE_DEBUG)
+    amqp_basic_deliver_t *d;
+    amqp_basic_properties_t *p;
+#endif
+    size_t body_target;
+    size_t body_received;
+
+    amqp_basic_consume(*conn, channel, callback_queue, amqp_empty_bytes, 0, 1,
+                       0, amqp_empty_table);
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Set the callback queue"))
+        return false;
+
+    /*Wait for three frames: AMQP_FRAME_METHOD, AMQP_FRAME_HEADER,
+     * AMQP_FRAME_BODY*/
+    for (;;) {
+        amqp_maybe_release_buffers(*conn);
+        amqp_simple_wait_frame(*conn, &frame);
+        if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Wait method frame"))
+            return false;
+
+        ddprintf(MSG_PREFIX "Frame type: %u channel: %u\n", frame.frame_type, frame.channel);
+
+        if (frame.frame_type != AMQP_FRAME_METHOD)
+            continue;
+
+        ddprintf(MSG_PREFIX "Method: %s\n", amqp_method_name(frame.payload.method.id));
+
+        if (frame.payload.method.id != AMQP_BASIC_DELIVER_METHOD)
+            continue;
+
+#if defined(ENABLE_DEBUG)
+        d = (amqp_basic_deliver_t *) frame.payload.method.decoded;
+        ddprintf(MSG_PREFIX "Delivery: %u exchange: %.*s routingkey: %.*s\n",
+               (unsigned) d->delivery_tag, (int) d->exchange.len,
+               (char *) d->exchange.bytes, (int) d->routing_key.len,
+               (char *) d->routing_key.bytes);
+#endif
+
+        amqp_maybe_release_buffers(*conn);
+        amqp_simple_wait_frame(*conn, &frame);
+        if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Wait header frame"))
+            return false;
+
+        if (frame.frame_type != AMQP_FRAME_HEADER) {
+            ddprintf("Unexpected header!");
+            return false;
+        }
+
+#if defined(ENABLE_DEBUG)
+        p = (amqp_basic_properties_t *) frame.payload.properties.decoded;
+        if (p->_flags & AMQP_BASIC_CONTENT_TYPE_FLAG) {
+            ddprintf(MSG_PREFIX "Content-type: %.*s\n", (int) p->content_type.len,
```

Comment:
Ditto.

Reply:
`p` is defined and used by `ddprintf` when `make BUILD_DEBUG`. Therefore, the `ENABLE_DEBUG` needs to be used.